net/relay.go

```diff
@@ -31,27 +31,51 @@ func (h *Host) DiscoverRelayers() ([]peer.ID, error) {
 func (h *Host) handleRelayStream(stream libp2pnetwork.Stream) {
 	defer func() { _ = stream.Close() }()
 
-	// TODO: If the request is from a Maker/OfferID combo that we did a swap with, we
-	//       should always be willing to relay.
-	if !h.isRelayer {
-		return
-	}
-
 	msg, err := readStreamMessage(stream, maxRelayMessageSize)
 	if err != nil {
 		log.Debugf("error reading RelayClaimRequest: %s", err)
 		return
 	}
 
+	curPeer := stream.Conn().RemotePeer()
+
 	req, ok := msg.(*RelayClaimRequest)
 	if !ok {
-		log.Debugf("ignoring wrong message type=%s sent to relay stream", message.TypeToString(msg.Type()))
+		log.Debugf("ignoring wrong message type=%s sent to relay stream from %s",
+			message.TypeToString(msg.Type()), curPeer)
+		return
+	}
+
+	// Handle case where we are not a relayer, and the request didn't set the offerID
+	// to indicate that it make from a running swap partner.
+
+	// While HandleRelayClaimRequest(...) will do lower level validation on the
+	// claim request, there are 2 validations best handled here:
+	// (1) If the network layer is not advertising that we are a relayer to the
+	//     DHT, we should not be getting claim requests targeted for open
+	//     relayers (i.e. requests that do not have the OfferID set).
+	// (2) If the request is purportedly from a maker to a taker of a current
+	//     swap, then:
+	//     (a) The swap should exist in our swaps map
+	//     (b) The peerID who sent us the request much match the peerID with
+	//         whom we are performing the swap.
```

Comment:
do we validate that the offerID and contractSwapID in the relay request match? (more of a note to self while reviewing)

Reply:
upon reviewing I don't think this is checked unless I missed it, pls add

Reply:
okay it's in `HandleRelayClaimRequest` I believe

Reply:
Yes. I did it there, because the backend has access to the database. The checks here are only network/peer related checks.